/* Copyright 2022 The TensorFlow Authors. All Rights Reserved.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
==============================================================================*/

syntax = "proto3";

package xla.gpu;

import "tensorflow/compiler/xla/service/hlo.proto";

message XlaRuntimeGpuExecutableProto {
  message ConstantInfoProto {
    string symbol_name = 1;
    bytes content = 2;
    int32 allocation_index = 3;
  }

  XlaRuntimeExecutableProto xla_runtime_executable = 1;

  // XLA-specific attributes of the executable's entry function.
  EntryFunctionAttributes entry_func_attrs = 2;

  // PTX for the compiled GPU kernels.
  string gpu_asm_text = 3;

  // Corresponding CUBIN for the above PTX.
  bytes gpu_binary = 4;

  // Constants required by the serialized executable.
  repeated ConstantInfoProto constants = 5;
}
